clang/include/clang/CConv/ConstraintVariables.h

```diff
@@ -377,7 +377,7 @@ class PointerVariableConstraint : public ConstraintVariable {
   // Get the set of constraint variables corresponding to the arguments.
   const std::set<ConstraintVariable *> &getArgumentConstraints() const;
 
-  ConstraintVariable *getCopy(Constraints &CS) override;
+  PointerVariableConstraint *getCopy(Constraints &CS) override;
```

Comment:
So in C++ you can override a method whose return type is `T` with a method whose return type is `S <: T` ?

Reply:
I wasn't sure of this myself, but it does seem to be the case. The `override` annotation will trigger an error if try you an invalid override but, it doesn't complain here.